client/gutenberg/extensions/email-subscribe/index.js

```diff
@@ -0,0 +1,37 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { __ } from 'gutenberg/extensions/presets/jetpack/utils/i18n';
+import EmailSubscribeEdit from './edit';
+import './editor.scss';
+
+export const name = 'email-subscribe';
+
+export const settings = {
+	title: __( 'Email Subscribe', 'jetpack' ),
+	icon: 'email',
+	category: 'widgets',
+	keywords: [ __( 'email', 'jetpack' ), __( 'mailchimp', 'jetpack' ), 'jetpack' ],
+
+	edit: EmailSubscribeEdit,
+
+	// We will generate shortcode as a fallback, so that this works even if gutenberg does not.
+	save: function( props ) {
+		let attrs = '';
+		if ( props.attributes ) {
+			attrs =
+				' ' +
+				Object.keys( props.attributes )
```

Comment:
As @simison suggested, we could bypass doing this by using `@wordpress/shortcode`, but in case you don't wish to use it, this sequence looks like a good use case for `_. transform()`